---
title: C28198
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28198
helpviewer_keywords:
- C28198
ms.assetid: 8bad4acb-712c-43f5-81d1-45d92092d4c5
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 78988ce7995e3ae5fde0a6391cc0cf7b46128362
ms.sourcegitcommit: 485ffaedb1ade71490f11cf05962add1718945cc
ms.translationtype: MT
ms.contentlocale: pl-PL
ms.lasthandoff: 10/16/2019
ms.locfileid: "72448306"
---
# <a name="c28198"></a>C28198
Ostrzeżenie C28198: prawdopodobnie wyciek pamięci z powodu wyjątku.

 To ostrzeżenie wskazuje, że przydzieloną pamięć nie jest zwalniana po wystąpieniu wyjątku. Instrukcja na końcu ścieżki może zgłosić wyjątek. Pamięć została przeniesiona do funkcji, która mogła zostać później zwolniona o kopię.

 To ostrzeżenie jest bardzo podobne do ostrzeżenia [C28197](../code-quality/c28197.md). W tym miejscu można również użyć adnotacji, które są zalecane do użycia z ostrzeżeniem [C28197](../code-quality/c28197.md) .

## <a name="example"></a>Przykład
 Poniższy przykład kodu generuje to ostrzeżenie:

```
char *p1 = new char[10];
char *p2 = new char[10];

test(p1);   // does not save a copy of p

delete[] p2;
delete[] p1;
```

 Poniższy przykład kodu pozwala uniknąć tego ostrzeżenia:

```
char *p1 = new char[10];
char *p2 = NULL;

test(p1);   // does not save a copy of p
try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
```
